Add the filesystem offload doc (#5762)
apache/pulsar: site2/docs/cookbooks-tiered-storage.md

We recommend you add a life cycle rule your bucket to expire incomplete multi-part upload after a day or two to avoid
getting charged for incomplete uploads.

## The Filesystem offloading mechanism

A topic data was stored with a lot of ledgers. Offload these ledgers into filesystem, we need a path for it.

> The implementation of the Filesystem offload to generate the path for **tenant/namespace/topic/ledgerId**.

Reviewer: The path of the data is `basePath` +  /tenant/namespace/domain/topic/ +  ledgerId-UUID right?

Author: yes, it should be in this way.

In both cases, these should not be touched unless you know what you are doing.

### "filesystem" Driver configuration

#### Configure connection address

> 

Reviewer 2: ```suggestion
You can configure the connection address in the `broker.conf` file.
```

```conf
fileSystemURI="hdfs://127.0.0.1:9000"
```
#### Configure Hadoop profile path

> In the configuration file under this path, you can configure the base path and authentication and so on.

Reviewer 2: ```suggestion
The configuration file is stored in the Hadoop profile path. It contains various settings, such as base path, authentication, and so on.
```

```conf
fileSystemProfilePath="../conf/filesystem_offload_core_site.xml"
```

> The model for storing this topic data use `org.apache.hadoop.io.MapFile`, so we can use all of the configuration in Apache Hadoop for `org.apache.hadoop.io.MapFile`, the following example :

Reviewer 2: ```suggestion
The model for storing topic data uses `org.apache.hadoop.io.MapFile`. You can use all of the configurations in `org.apache.hadoop.io.MapFile` for Hadoop.

**Example**
```

```conf

    <property>
        <name>fs.defaultFS</name>
        <value></value>
    </property>
    
    <property>
        <name>hadoop.tmp.dir</name>
        <value>pulsar</value>
    </property>
    
    <property>
        <name>io.file.buffer.size</name>
        <value>4096</value>
    </property>
    
    <property>
        <name>io.seqfile.compress.blocksize</name>
        <value>1000000</value>
    </property>
    <property>
    
        <name>io.seqfile.compression.type</name>
        <value>BLOCK</value>
    </property>
    
    <property>
        <name>io.map.index.interval</name>
        <value>128</value>
    </property>
    
```

> A detailed explanation of the role of these configurations and other configurations can be found in [Filesystem Storage](http://hadoop.apache.org/).

Reviewer 2: ```suggestion
For more information about the configurations in `org.apache.hadoop.io.MapFile`, see [Filesystem Storage](http://hadoop.apache.org/).
```

In both cases, these should not be touched unless you know what you are doing.

### "filesystem" Driver configuration

#### Configure connection address

You can configure the connection address in the `broker.conf` file.

```conf
fileSystemURI="hdfs://127.0.0.1:9000"
```
#### Configure Hadoop profile path

The configuration file is stored in the Hadoop profile path. It contains various settings, such as base path, authentication, and so on.

```conf
fileSystemProfilePath="../conf/filesystem_offload_core_site.xml"
```

The model for storing topic data uses `org.apache.hadoop.io.MapFile`. You can use all of the configurations in `org.apache.hadoop.io.MapFile` for Hadoop.

> **Example**

Reviewer 2: ```suggestion

**Example**
```

Add a blank line to correct this issue.
![image](https://user-images.githubusercontent.com/50226895/69925968-6145f200-14ed-11ea-92ff-21cb097a488e.png)
